liblangutil/Scanner.h

```diff
@@ -248,6 +253,19 @@ class Scanner
 	/// Scans a slash '/' and depending on the characters returns the appropriate token
 	Token scanSlash();
 
+	/// Tries scanning given octet sequence and advances reading position respectively iff found.
+	/// @returns true if it could be scanned, false otherwise.
+	bool tryScanByteSequence(std::string_view _sequence)
+	{
+		if (!m_source->prefixMatch(_sequence))
+			return false;
+
+		for (size_t i = 0; i < _sequence.size(); ++i)
+			advance();
+
+		return true;
```

Comment:
I'd add `assert(!isSourcePastEndOfInput())` here.

Reply:
if the prefix matches, you are already assured that you have size(prefix) bytes that you can consume. hmm....

Reply:
Only if `prefixMatch()` and `advance()` do what they claim and have no bugs. It's not strictly necessary here, assertions never are, but I usually add a lot of them in my code to document and verify assumptions like this.

Reply:
I just checked. advance() is actually bailing out early if it cannot advance.  I added an early-exit guard at the top of this function now though, too.